lib/active_storage/service/open_stack_service.rb

```diff
@@ -135,6 +129,23 @@ def change_content_type(key, content_type)
     end
 
   private
+    def private_url(key, expires_in:, filename:, disposition:, content_type:, **)
+      expire_at = unix_timestamp_expires_at(expires_in)
+      generated_url = client.create_temp_url(container, key, expire_at, 'GET', filename: filename)
+      generated_url += '&inline' if disposition.to_s != 'attachment'
+      # unfortunately OpenStack Swift cannot overwrite the content type of an object via a temp url
+      # so we just ignore the content_type argument here
+
+      generated_url
+    end
+
+    def public_url(key, **options)
+      uri = URI.parse(private_url(key, **options))
```

Comment:
You can use [client.public_url(container, key)](https://github.com/fog/fog-openstack/blob/c51856a2eb2567b421656598fb519ecaa2cf4969/lib/fog/openstack/storage/requests/public_url.rb)